### ISSUE TITLE
PHP 8.1 support

### ISSUE THREAD

**Issue by peterjaap** [prismicio-community/php-kit#182]
On PHP 8.1, I'm getting this notice;

> Deprecated: Optional parameter $name declared before required parameter $fields is implicitly treated as a required parameter in /data/client/magento2/vendor/prismic/php-sdk/src/Prismic/Form.php on line 65

Afaik there's nothing stopping us from changing the constructor to;

```
diff --git src/Prismic/Form.php src/Prismic/Form.php
index 7de168d..5f5c027 100644
--- src/Prismic/Form.php
+++ src/Prismic/Form.php
@@ -63,12 +63,12 @@ class Form
      * @param array  $fields    the list of Prismic::FieldForm objects that can be used
      */
     private function __construct(
-        ?string $name = null,
-        string  $method,
-        ?string $rel = null,
-        string  $enctype,
-        string  $action,
-        array   $fields
+        string $name = null,
+        string $method,
+        string $rel = null,
+        string $enctype,
+        string $action,
+        array  $fields
     ) {
         $this->name    = $name;
         $this->method  = $method;

```

**Comment by peterjaap** [comment]
Here's another one;

> Exception #0 (Exception): Deprecated Functionality: htmlentities(): Passing null to parameter #2 ($flags) of type int is deprecated in /data/client/magento2/vendor/prismic/php-sdk/src/Prismic/Dom/RichText.php on line 154

**Comment by c0nst4ntin** [comment]
@peterjaap I ran into similar problems on some client projects and therefore looked into this topic a little bit more. I created a new pull-request (#190) that contains similar changes to #183 but also solves other challenges such as making PHP 8 a minimum requirement and updating the test suite to work with the new version of PHP.

**Comment by peterjaap** [comment]
Looks like Prismic has completely forgotten or neglected PHP. Maybe @srenault can help us out by getting Prismic to give the PHP community some love?

**Comment by c0nst4ntin** [comment]
@peterjaap I was in contact with some of the developers at Prismic previously. I'll try to address this and see if we can get this repository back to its old glory. Maybe even improve some of the formatting and code quality

**Comment by pTinosq** [comment]
Any updates on this? Really praying for 8.1 support soon 😥

**Comment by peterjaap** [comment]
@pTinosq we've given up on buy-in from Prismic. They're a JS crowd, they don't like PHP. You can use our fork instead https://github.com/elgentos/prismicio-php-kit

**Comment by pTinosq** [comment]
> @pTinosq we've given up on buy-in from Prismic. They're a JS crowd, they don't like PHP. You can use our fork instead https://github.com/elgentos/prismicio-php-kit

Oh wow that's really annoying. Should I use a specific fork or is the elegentos version sufficient?

**Comment by peterjaap** [comment]
@pTinosq the elgentos one is most likely sufficient if you're looking PHP 8.1 compat

**Comment by pTinosq** [comment]
Ah that's good. I'll keep my notifications on for the PR. Hopefully Prismic can pull through 🙏🙏

**Comment by c0nst4ntin** [comment]
@peterjaap @pTinosq I'm happy to tell you, that this issue was fixed in **[feat!: support PHP 8 #190](https://github.com/prismicio-community/php-kit/pull/190)**

To make use of this change please upgrade to **[Version 5.2.0](https://github.com/prismicio-community/php-kit/releases/tag/5.2.0)**

I will proceed to close this issue.

**Comment by pTinosq** [comment]
@c0nst4ntin This is the news I needed to wake up to. Fantastic stuff 🎉🎉

**Comment by jonathan-bird** [comment]
@c0nst4ntin This does not look to be (completely) fixed in the new 5.2 release. 

I am running PHP 8.1 and seeing lots of `htmlentities(): Passing null to parameter #2 ($flags) of type int is deprecated in /prismic/php-sdk/src/Prismic/Dom/RichText.php on line 154` errors.
